### PR TITLE
Sangeetparashar/yld 19 display current stake and balance on stakebtc

### PR DESCRIPTION
**Pull request by sangeetparashar**
* Add axios and query provider to make api requests
* imported and updated multiple interfaces + update the /stake/btc page
* Added and formatted table to improve the look and feel of the table

https://github.com/user-attachments/assets/7df7d35c-bdde-40f7-a88f-b898a24c662e

### CODE REVIEW

File: src/app/stake/btc/page.tsx [hunk 1]
```diff
@@ -1,86 +1,192 @@
...
-              <Table.Cell className="px-4 py-2">
-                {item.totalDelegation}
-              </Table.Cell>
-              <Table.Cell className="px-4 py-2">{item.commission}</Table.Cell>
-            </Table.Row>
-          ))}
-        </Table.Body>
-      </Table.Root>
-    </div>
+    <Theme
```
Comment (MattNguyen) [inline src/app/stake/btc/page.tsx]: shouldn't this be in the layout?

File: src/app/stake/btc/page.tsx [hunk 2]
```diff
@@ -1,86 +1,192 @@
 "use client";
```
Comment (MattNguyen) [inline src/app/stake/btc/page.tsx:1]: non-blocking: my preference would be that we create a server component here that fetches the first page of finality providers, which gets passed into the client component.

```
async function Page() {
  try {
      const finalityProviders = await getFinalityProviders({ page: 1 });
      return <StakeBtc providers={finalityProviders} />;
  } catch() { return <Error /> }
}
```

`providers` would be set as initialData in useInfiniteQuery and initialPageParams would be set to page=1

File: src/app/stake/btc/page.tsx [hunk 3]
```diff
@@ -1,86 +1,192 @@
...
+      radius="full"
+    >
+      <div className="flex flex-col items-center justify-center min-h-screen px-4 sm:px-8">
+        <h1 className="text-4xl font-bold mb-8 text-center">
+          Select Finality Provider
+        </h1>
+        <div className="w-full max-w-5xl">
+          <div className="relative overflow-x-auto rounded-lg shadow-lg bg-white">
+            <Table.Root variant="surface" size="1" layout="auto">
+              <Table.Header className="sticky top-0 bg-gray-200 z-10">
```
Comment (MattNguyen) [inline src/app/stake/btc/page.tsx]: i don't think `sticky` is doing anything. See my vercel comment about making the table itself scrollable inside a ScrollArea. There shouldn't be page scrolling (or at least there isn't a need for page scrolling yet).

File: src/app/stake/btc/page.tsx [hunk 4]
```diff
@@ -1,86 +1,192 @@
...
+                <Table.Row className="bg-gray-200">
+                  <Table.ColumnHeaderCell className="px-4 py-2">
```
Comment (MattNguyen) [inline src/app/stake/btc/page.tsx]: I think ColumnHeaderCell has props to define padding - https://www.radix-ui.com/themes/docs/components/table#size

**Review by MattNguyen — APPROVED** [review]
Just a few comments, but nothing blocking. Addressing feedback can happen in a separate PR if you wish.

**Comment by sangeetparashar** [comment]
Rest of the PR comments will be addressed in the next iteration

File: src/app/stake/btc/page.tsx (hunk 1 above)
Comment (sangeetparashar) [reply src/app/stake/btc/page.tsx]: removed it. The main issue was global.css had a parent level black bg, for I was adding themes per page. It's not required anymore

File: src/app/stake/btc/page.tsx (hunk 3 above)
Comment (sangeetparashar) [reply src/app/stake/btc/page.tsx]: removed it

**Pull request merged**